sycl/source/detail/scheduler/scheduler.hpp

```diff
@@ -711,22 +727,22 @@ class Scheduler {
     static std::vector<EventImplPtr> getWaitList(EventImplPtr Event);
 
     /// Waits for the command, associated with Event passed, is completed.
-    static void waitForEvent(EventImplPtr Event);
+    /// \param GraphReadLock read-lock which is already acquired for reading
+    static void waitForEvent(EventImplPtr Event, ReadLockT &GraphReadLock);
 
     /// Enqueues the command and all its dependencies.
     ///
     /// \param EnqueueResult is set to specific status if enqueue failed.
+    /// \param GraphReadLock read-lock which is already acquired for reading
     /// \return true if the command is successfully enqueued.
     static bool enqueueCommand(Command *Cmd, EnqueueResultT &EnqueueResult,
                                BlockingT Blocking = NON_BLOCKING);
   };
 
-  void waitForRecordToFinish(MemObjRecord *Record);
+  void waitForRecordToFinish(MemObjRecord *Record, ReadLockT &GraphReadLock);
```

Comment:
Minor comment: would you please add a comment to this function too?

Reply:
Added